OpenRA.Game/Map/MapCache.cs

```diff
@@ -122,6 +122,47 @@ public void LoadMaps()
 			}
 		}
 
+		public IEnumerable<Map> EnumerateMapsWithoutCaching()
+		{
+			// Utility mod that does not support maps
+			if (!modData.Manifest.Contains<MapGrid>())
+				yield break;
+
+			// Enumerate map directories
+			foreach (var kv in modData.Manifest.MapFolders)
+			{
+				var name = kv.Key;
+				var classification = string.IsNullOrEmpty(kv.Value)
+					? MapClassification.Unknown : Enum<MapClassification>.Parse(kv.Value);
```

Comment:
`Enum.TryParse` and list the valid values upon failure.

Reply:
Ignore this I suppose. You're just moving code afterall.

Reply:
This would be a good opportunity to fix the original code (since this is pulling out a common helper).